examples/powertools-examples-core/kotlin/src/main/kotlin/helloworld/App.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package helloworld
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.RequestHandler
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent
+import com.amazonaws.xray.entities.Subsegment
+import org.apache.logging.log4j.LogManager
+import software.amazon.cloudwatchlogs.emf.logger.MetricsLogger
+import software.amazon.cloudwatchlogs.emf.model.DimensionSet
+import software.amazon.cloudwatchlogs.emf.model.Unit
+import software.amazon.lambda.powertools.logging.Logging
+import software.amazon.lambda.powertools.logging.LoggingUtils
+import software.amazon.lambda.powertools.metrics.Metrics
+import software.amazon.lambda.powertools.metrics.MetricsUtils
+import software.amazon.lambda.powertools.tracing.CaptureMode
+import software.amazon.lambda.powertools.tracing.Tracing
+import software.amazon.lambda.powertools.tracing.TracingUtils
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.net.URL
+import java.util.stream.Collectors
+
+/**
+ * Handler for requests to Lambda function.
+ */
+
+class App : RequestHandler<APIGatewayProxyRequestEvent?, APIGatewayProxyResponseEvent> {
```

Comment:
@AlexeySoshin you are our _idiomatic kotlin enthusiast_ - do you want to go through this again, or do you have someone else we can use for an extra set of eyes?

Reply:
I'm happy to go through this again, but I'm not sure how much effort we want to put into it.
Since the underlying library is in Java, there's an effort to make it "nice".

Take this line for example:
https://github.com/aws-powertools/powertools-lambda-java/pull/1454/files#diff-15c9fc29d29ca3087fd0dd942b8d41046bcf02ec0bc828c3c702a6030f72d710R61

In Kotlin, idiomatic way would be to use 
```kotlin
 TracingUtils["Test"] = "New"
```
Or
```kotlin
TracingUtils.annotations["Test"] = "New"
```
But in order to have that, we need to introduce some extension functions in that project.

Reply:
For an initial step, I think we should make it as nice as possible without having to make changes to the core module. If we see subsequent demand for Kotlin we can look at extending the PT components themselves to make it jazzier. WDYT @AlexeySoshin ?

Reply:
agree, let's keep PT v1 like this, and potentially think about this in v2 (part of the interface review)